### PR TITLE
impl(otel): traced async backoffs

### PR DESCRIPTION
**Pull request by dbolduc**
Part of the work for #10618 

The analog of #10589, but async.

I will incorporate it in the `AsyncRetryLoop`s and `AsyncPollingLoop`s in separate PRs.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11048)

### CODE REVIEW

File: google/cloud/internal/grpc_opentelemetry.h [hunk 1]
```diff
@@ -93,6 +96,20 @@ future<T> EndSpan(
...
+ */
+template <typename Rep, typename Period>
+future<StatusOr<std::chrono::system_clock::time_point>> TracedAsyncBackoff(
+    CompletionQueue& cq, std::chrono::duration<Rep, Period> duration) {
+#ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
+  if (TracingEnabled(CurrentOptions())) {
+    return EndSpan(MakeSpan("Async Backoff"), cq.MakeRelativeTimer(duration));
+  }
+#endif  // GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
+  return cq.MakeRelativeTimer(duration);
```
Comment (devbww) [inline google/cloud/internal/grpc_opentelemetry.h:110]: I would be tempted to factor out the common `MakeRelativeTimer()` bits.

**Review by devbww — APPROVED** [review]

File: google/cloud/internal/grpc_opentelemetry.h (hunk 1 above)
Comment (dbolduc) [reply google/cloud/internal/grpc_opentelemetry.h]: Done, but you cost me a line of code.

Comment (devbww) [reply google/cloud/internal/grpc_opentelemetry.h]: So then maybe that's timer = EndSpan(MakeSpan("Async Backoff"), std::move(timer));?

Comment (dbolduc) [reply google/cloud/internal/grpc_opentelemetry.h]: Done.

**Pull request merged**